packages/dara-core/dara/core/main.py

```diff
@@ -423,3 +429,40 @@ def start(extra=None):
     config._run_discovery(config_module)
 
     return _start_application(config=config._to_configuration())
+
+
+class CustomResponse(Response):
+    media_type = 'application/json'
+
+    def __init__(
+        self,
+        content: typing.Any,
+        status_code: int = 200,
+        headers: typing.Optional[typing.Dict[str, str]] = None,
+        media_type: typing.Optional[str] = None,
+        background: typing.Optional[BackgroundTask] = None,
+    ) -> None:
+        super().__init__(content, status_code, headers, media_type, background)
+
+    def render(self, content: typing.Any) -> bytes:
+        import math
+
+        custom_encoder = {}
+        custom_encoder[float] = lambda x: None if math.isnan(x) or math.isinf(x) else x
```

Comment:
Let's move these out of `render` to the top-level of the file, doesn't need to be redefined and reimported within `render` each time, as this is ran for every request this can affect performance.